xenmgr/Vm/Config.hs

```diff
@@ -683,19 +682,33 @@ diskSpec :: Uuid -> Disk -> Rpc DiskSpec
 diskSpec uuid d  = do
   stubdom <- readConfigPropertyDef uuid vmStubdom False
   hd_type <- readConfigPropertyDef uuid vmHdType "ide"
-  return $ printf "'%s,%s,%s,%s,%s,%s'"
-             (diskPath d) (fileToRaw (enumMarshall $ diskType d)) (cdType stubdom d) (adjDiskDevice d hd_type) (enumMarshall $ diskMode d) (if ((enumMarshall $ diskDeviceType d) == "cdrom") then (enumMarshall $ diskDeviceType d) else "")
+  return $ printf "'format=%s,backendtype=%s,vdev=%s,access=%s,devtype=%s,script=%s,target=%s'"
+             (fileToRaw disk_type)
+             (cdType stubdom d)
+             (adjDiskDevice d hd_type)
+             (enumMarshall $ diskMode d)
+             (if (disk_dev_type == "cdrom") then disk_dev_type else "disk")
+             (script disk_type)
+             (target disk_type (diskPath d))
   where
+    disk_type = enumMarshall $ diskType d
+    disk_dev_type = enumMarshall $ diskDeviceType d
     cdType stubdom d =
-      case (enumMarshall $ diskDeviceType d) of
-          "cdrom" -> if stubdom then "backendtype=tap" else "backendtype=phy"
-          _       -> if (enumMarshall $ diskType d) == "phy" then "backendtype=phy" else "backendtype=tap"
-    fileToRaw typ = if typ == "file" || typ == "phy" then "raw" else typ
+      case disk_dev_type of
+          "cdrom" -> "phy"
+          _       -> "phy"
+    fileToRaw typ = "raw"
```

Comment:
```qcow``` is deprecated.
```qed``` improves upon ```qcow2``` so ```qcow2``` is effectively deprecated I would think.

```vhd``` still being supported even though blocktap has been removed from libxl is a head scratcher for me?  Idk what it's purpose is at this point.

Reply:
vhds are supported by QEMU as "vpc".  It doesn't suit our purposes though since QEMU doesn't have encrypted vhd support.

Oh, I looked and QED is deprecated.  There is qcow3, and it was integrated into a unified qcow2 driver in QEMU.
https://wiki.qemu.org/Features/QED
https://wiki.qemu.org/Features/Qcow3